lib/manageiq/appliance_console/database_admin.rb

```diff
@@ -88,29 +89,56 @@ def ask_smb_file_options
       end
 
       def ask_to_delete_backup_after_restore
-        if backup_type == LOCAL_FILE
+        if action == :restore && backup_type == LOCAL_FILE
           say("The local database restore file is located at: '#{uri}'.\n")
           @delete_agree = agree("Should this file be deleted after completing the restore? (Y/N): ")
         end
       end
 
       def confirm_and_execute
-        say("\nNote: A database restore cannot be undone.  The restore will use the file: #{uri}.\n")
-        if agree("Are you sure you would like to restore the database? (Y/N): ")
-          say("\nRestoring the database...")
-          rake_success = ManageIQ::ApplianceConsole::Utilities.rake(task, task_params)
-          if rake_success && delete_agree
-            say("\nRemoving the database restore file #{uri}...")
-            File.delete(uri)
-          elsif !rake_success
-            say("\nDatabase restore failed. Check the logs for more information")
-          end
+        if allowed_to_execute?
+          processing_message
+          run_rake
         end
         press_any_key
       end
 
+      def allowed_to_execute?
+        if action == :restore
+          say("\nNote: A database restore cannot be undone.  The restore will use the file: #{uri}.\n")
+          agree("Are you sure you would like to restore the database? (Y/N): ")
+        else
+          true
+        end
+      end
+
+      def file_menu_args
+        [
+          action == :restore ? "Restore Database File" : "Backup File Location",
+          FILE_OPTIONS,
+          LOCAL_FILE,
+          nil
+        ]
+      end
+
       def setting_header
-        say("#{I18n.t("advanced_settings.dbrestore")}\n\n")
+        say("#{I18n.t("advanced_settings.db#{action}")}\n\n")
+      end
+
+      private
+
+      def processing_message
+         say("\n#{action == :restore ? "Restoring the database" : "Running Database backup to #{uri}"}...")
```

Comment:
This looks like trash, but I think when I end up adding the `:dump` action into the mix, I will end up making this look a little nicer then.

Reply:
Yeah, this is fine for now :+1: